### ISSUE TITLE
Comments should be indented

### ISSUE THREAD

**Issue by technic** [jupyterlab/jupyterlab#6957]
**Describe the ~~bug~~ feature**
JupyterLab toggle comment action should create indented comments i.e.
```python
def f(x):
    # comment
    return x**2
```
instead of 
```python
def f(x):
#    comment
    return x**2
```

This was discussed before for the notebook https://github.com/jupyter/notebook/issues/954, but I didn't find any info regarding JupyterLab.

**To Reproduce**
Type some code, press `Ctrl`+`/` button

**Expected behavior**
Correct indentation like almost any IDE or text editor will do.

**Desktop (please complete the following information):**

- OS: [Windows 10]
- Browser [chrome 75.0]
- JupyterLab [Version 1.0.2]

**Comment by lukelbd** [comment]
Hoping someone has time to look into this soon. The current block-commenting behavior results in code that does not conform to the increasingly popular ["black" format](https://github.com/psf/black), and therefore conflicts with the [code-formatter plugin](https://jupyterlab-code-formatter.readthedocs.io/en/latest/).

For example, suppose I have the following python cell: 

```python
if condition:
    do_something_0
    do_something_1
    do_something_2
    do_something_3
```

and comment out code with `Cmd + /` as follows:

```python
if condition:
    do_something_0
#     do_something_1
#     do_something_2
    do_something_3
```

If I then automatically format the cell with `black` (which is the *default* formatter for the [code-formatter extension](https://jupyterlab-code-formatter.readthedocs.io/en/latest/)), the code looks like this:

```python
if condition:
    do_something_0
    #     do_something_1
    #     do_something_2
    do_something_3
```

and uncommenting the code with `Cmd + /` again results in invalid python code:

```python
if condition:
    do_something_0
        do_something_1
        do_something_2
    do_something_3
```

Plus, the current `Cmd + /` block-commenting behavior just looks plain strange to me.

If it helps, it looks like this issue was addressed for jupyter notebooks in jupyter/notebook#954.

**Comment by lukelbd** [comment]
@blink1073 Who would be best equipped to address this? This issue isn't critical but has been a year with no activity, not sure if it's on anyone's radar. Was just hoping to get acknowledgement from the core jupyter contributors.

**Comment by technic** [comment]
I would like to have this fixed. Please let us know if you need any help with it.

**Comment by jasongrout** [comment]
Playing around with this in VS Code, it seems that for python at least, they indent to the minimum level of the non-whitespace lines in the selection. So look at all the selected lines, filter out any that are only whitespace, then calculate the minimum indentation of lines that are left, and insert the line comment character at that indentation level.

If someone wants to work on this, please go for it!

Here is where we are assigning that functionality to the shortcut: https://github.com/jupyterlab/jupyterlab/blob/abc0dd4255a037af37828122bd2e199857fd64db/packages/codemirror/src/factory.ts#L24-L25

We use the codemirror comment addon to provide that function:

https://github.com/jupyterlab/jupyterlab/blob/abc0dd4255a037af37828122bd2e199857fd64db/packages/codemirror/src/editor.ts#L30

Here is the definition of that CodeMirror addon function to toggle comments: https://github.com/codemirror/CodeMirror/blob/master/addon/comment/comment.js

So one could either modify the codemirror addon to have this feature, or we could pull that function into JupyterLab with the desired modifications.

**Comment by jasongrout** [comment]
Actually, it looks like that codemirror command supports an 'indent' option to try to align the comment with the current indentation. See https://codemirror.net/doc/manual.html#addons (search for `comment.js`). So perhaps we just need to call the commenting command with the right options?

**Comment by jasongrout** [comment]
It looks like the `indent` option only aligns to the first line, not the minimum indent across all lines, so it's not quite the same as vs code, but maybe that's good enough for now? Again, PRs are welcome!

**Comment by fcollonval** [comment]
Fixed by #11063

**Comment by chrisjsewell** [comment]
Ah brilliant thanks, this is such a quality of life improvement 🙏

**Comment by jasongrout** [comment]
Fixed by #11063 in 3.1.x
